kubetest/extract.go

```diff
@@ -483,3 +483,230 @@ func saveState(save string) error {
 	}
 	return nil
 }
+
+type extractFederationStrategy struct {
```

Comment:
the entire federation extract should be in it's own file, maybe rename extract.go -> extract_k8s.go, and add a extract_federation.go

Reply:
Done